modules/friendlytag.js

```diff
@@ -2009,47 +2035,97 @@ Twinkle.tag.callback.evaluate = function friendlytagCallbackEvaluate(e) {
 					return;
 				}
 			}
-			if ((params.tags.indexOf('Not English') !== -1) && (params.tags.indexOf('Rough translation') !== -1)) {
-				alert('Please select only one of {{not English}} and {{rough translation}}.');
+
+			if (checkIncompatible(['Not English', 'Rough translation'])) {
 				return;
 			}
-			if (params.tags.indexOf('History merge') !== -1 && params.histmergeOriginalPage.trim() === '') {
-				alert('You must specify a page to be merged for the {{history merge}} tag.');
+			if (checkParameter('History merge', 'histmergeOriginalPage', 'a page to be merged')) {
 				return;
 			}
-			if (params.tags.indexOf('Cleanup') !== -1 && params.cleanup.trim() === '') {
-				alert('You must specify a reason for the {{cleanup}} tag.');
+			if (checkParameter('Cleanup', 'cleanup')) {
 				return;
 			}
-			if (params.tags.indexOf('Expand language') !== -1 && params.expandLanguageLangCode.trim() === '') {
-				alert('You must specify language code for the {{expand language}} tag.');
+			if (checkParameter('Expand language', 'expandLanguageLangCode', 'a language code')) {
 				return;
 			}
 			break;
 
 		case 'file':
+			if (checkIncompatible(['Bad GIF', 'Bad JPEG', 'Bad SVG', 'Bad format'])) {
+				return;
+			}
+			if (checkIncompatible(['Should be PNG', 'Should be SVG', 'Should be text'])) {
+				return;
+			}
+			if (checkIncompatible(['Bad SVG', 'Vector version available'])) {
+				return;
+			}
+			if (checkIncompatible(['Bad JPEG', 'Overcompressed JPEG'])) {
+				return;
+			}
+			if (checkIncompatible(['PNG version available', 'Vector version available'])) {
+				return;
+			}
+
+			var extension;
+			if ((extension = Morebits.pageNameNorm.match(/\.(\w+)$/)) !== null) {
+				extension = extension[1];
```

Comment:
Per a [discussion with the requester](https://en.wikipedia.org/wiki/User_talk:Smjg#Twinkle_pull_request_for_your_file_incompatibility_checks), it should be possible to use MIME types for jpg/gif/png, with a fallback to this for svgs.  Something stupid like `((extension = $('.mime-type').text()) && extension.split(/\//)[1]) || ((extension = Morebits.pageNameNorm.match(/\.(\w+)$/)) !== null) && (extension = extension[1])` works 😆

Reply:
To get the extension, we can use `mw.Title.newFromText(Morebits.pageNameNorm).getExtension()` as well.

Reply:
Done